### DISCUSSION TITLE
Make the {% image %} tag support other images known to django, not just wagtailimages

### DISCUSSION THREAD

**Discussion by davecranwell** [wagtail/wagtail#11392] [Ideas]
For example on a current project, users upload a photo to represent an event, but events are submitted by the public and not everyone will have a suitable photo, so a placeholder is displayed alongside their event instead.

That placeholder image is something the developer has had to create, but as event photos get displayed in numerous places, we need to create numerous versions of that placeholder to suit the placement, which is tedious.

It would be great if the `{% image %}` tag could be given a path to a static image, and automatically generate the images at the dimensions we specify.

**Comment by kaedroho** [comment]
This one would need some database changes as the `Rendition` model is currently linked with the `Image` model. A possible solution might be to instead, link renditions to image source files. This has the added benefit that you can create a rendition from anything without needing to create an `Image` first. It would also make creating custom image models simpler as well.

**Comment by davecranwell** [comment]
A cheaper option might be to have a _separate_ image tag for static images, hooking into the same resizing features, but if an update to the current image tag would be workable, that would be best for me.

**Comment by kaedroho** [comment]
I think that it would be good to cache the resulting image which is something that hooking into `Rendition` gives us.

**Comment by davecranwell** [comment]
I've discovered a new need for this today. I'll be developing a more extensive user profile in due course, to store more user preferences associated with how they use the wagtail admin. One of these new fields is a proper avatar. At the moment we use Gravatar, which is confusing for pretty much everyone since a Wordpress account became a requirement. 

I'll be falling back on Gravatar, but adding a "Welcome" screen to wagtail in which users are urged to upload an avatar to a regular django `FileField`. That avatar will need to be resized into various formats as the situation dictates. Given the amount of work that's gone into the image resizing system it would be churlish to create these avatar size formats any other way.

This is perhaps a slightly different request to the original issue, as previously I wanted to resize any arbitrary image found on disk. This time it's an image that django is already aware of to some degree.

**Comment by davecranwell** [comment]
On a more practical point, users may be integrating Wagtail with existing products and may find it tedious to have to install multiple thumbnailing tags. If wagtail could thumbnail anything, it could be a replacement.

**Comment by killarny** [comment]
Curious whether this is going anywhere, or has been forgotten about or replaced with other functionality that I'm not aware of?

I'm VERY new to wagtail, but already really wishing I could pass a static image path to the `{% image %}` tag so that I can resize user avatar images as needed in my templates.

**Comment by thibaudcolas** [comment]
I’ve converted this from issues to Discussions, as this currently feels more like a "wouldn’t it be neat if…" cool idea, than something that anyone is likely to work on. There’s pretty clear needs for this, but also it’s a lot of work, and there are workarounds for most scenarios I can think of. There are also lots of thumbnailing services outside Wagtail that are more flexible, for people who need the flexibility to manipulate images from arbitrary sources.
